cmake/LinkCudaLibraries.cmake

```diff
@@ -0,0 +1,33 @@
+# Link CUDA libraries to the given target, i.e.: `target_link_libraries(target <args>)`
+#
+# Additionally makes sure CUDA stub libs don't end up being in RPath
+#
+# Example: link_cuda_libraries(mytarget PRIVATE ${CUDA_LIBRARIES})
+function(link_cuda_libraries target)
```

Comment:
Function name feels a bit misleading. Consider renaming to something like `link_libraries_excluding_stubs` or something like that)

Reply:
Makes sense, yes. Probably even `target_link_libraries_exclude_cuda_stubs`?